File: 02/claude-opus-4.5/python/solution.py
with open("input.txt") as f:
    input_data = f.read().strip()

def is_invalid_id_part1(num: int) -> bool:
    s = str(num)
    if len(s) % 2 != 0:
        return False
    half = len(s) // 2
    return s[:half] == s[half:]

def is_invalid_id_part2(num: int) -> bool:
    s = str(num)
    # Check for patterns repeated at least twice
    for pattern_len in range(1, len(s) // 2 + 1):
        if len(s) % pattern_len != 0:
            continue
        pattern = s[:pattern_len]
        repetitions = len(s) // pattern_len
        if repetitions < 2:
            continue
        if pattern * repetitions == s:
            return True
    return False

# Part 1
part1 = 0
for range_str in input_data.split(','):
    if not range_str:
        continue
    start_str, end_str = range_str.split('-')
    start, end = int(start_str), int(end_str)
    for id_num in range(start, end + 1):
        if is_invalid_id_part1(id_num):
            part1 += id_num
print(f"Day 02 Part 1: {part1}")

# Part 2
part2 = 0
for range_str in input_data.split(','):
    if not range_str:
        continue
    start_str, end_str = range_str.split('-')
    start, end = int(start_str), int(end_str)
    for id_num in range(start, end + 1):
        if is_invalid_id_part2(id_num):
            part2 += id_num
print(f"Day 02 Part 2: {part2}")

File: 02/claude-opus-4.5/typescript/solution.ts
import * as fs from 'fs';

const input = fs.readFileSync('input.txt', 'utf-8').trim();

function isInvalidIdPart1(num: bigint): boolean {
    const s = num.toString();
    if (s.length % 2 !== 0) return false;
    const half = s.length / 2;
    return s.substring(0, half) === s.substring(half);
}

function isInvalidIdPart2(num: bigint): boolean {
    const s = num.toString();
    // Check for patterns repeated at least twice
    for (let patternLen = 1; patternLen <= Math.floor(s.length / 2); patternLen++) {
        if (s.length % patternLen !== 0) continue;
        const pattern = s.substring(0, patternLen);
        const repetitions = Math.floor(s.length / patternLen);
        if (repetitions < 2) continue;
        let isRepeated = true;
        for (let i = 1; i < repetitions; i++) {
            if (s.substring(i * patternLen, (i + 1) * patternLen) !== pattern) {
                isRepeated = false;
                break;
            }
        }
        if (isRepeated) return true;
    }
    return false;
}

// Part 1
let part1 = 0n;
const ranges = input.split(',').filter(r => r.length > 0);
for (const range of ranges) {
    const [startStr, endStr] = range.split('-');
    const start = BigInt(startStr);
    const end = BigInt(endStr);
    for (let id = start; id <= end; id++) {
        if (isInvalidIdPart1(id)) {
            part1 += id;
        }
    }
}
console.log(`Day 02 Part 1: ${part1}`);

// Part 2
let part2 = 0n;
for (const range of ranges) {
    const [startStr, endStr] = range.split('-');
    const start = BigInt(startStr);
    const end = BigInt(endStr);
    for (let id = start; id <= end; id++) {
        if (isInvalidIdPart2(id)) {
            part2 += id;
        }
    }
}
console.log(`Day 02 Part 2: ${part2}`);
